packages/bitcoin/src/fungible.ts

```diff
@@ -0,0 +1,113 @@
+import type { BitcoinResource } from "@buildwithsygma/core/src";
+import type { Network } from "bitcoinjs-lib";
+
+/* eslint-disable @typescript-eslint/no-unused-vars */
+
+/**
+ * @description Bitcoin transaction input template
+ * hash is the transaction hash of the previous unspent utxo
+ * index is the index of the previous unspent utxo in the vout object of the transaction
+ */
+type BaseInputTemplate = {
+  hash: string;
+  index: number;
+};
+
+/**
+ * @description Segwit input template to create bitcoin transaction
+ * witness uses the script and value of the previous unspent utxo. We take the script from the hex attribute in the scriptPubKey object
+ * script has to be hex encoded
+ * value is the amount of satoshis in the previous unspent utxo
+ */
+type WitnessInputTemplate = BaseInputTemplate & {
+  witness: {
+    script: Buffer;
+    value: number;
+  };
+};
+
+/**
+ * @description  output template to create bitcoin transaction
+ * evmDestinationAddress is encoded EVM address of the receiver
+ * resource is the bitcoin taproot address to send the transaction that will trigger briging the asset
+ * value is either satoshis to send or 0 for the encoded data
+ */
+type BaseOutputTemplate = {
+  evmDestinationAddress: string;
+  resource: BitcoinResource;
+  value: number;
+};
+
+/**
+ * @description Non-segwit input template to create bitcoin transaction
+ * nonWitnessUtxo is the raw transaction data of the previous unspent utxo hex encoded
+ */
+type NonWitnessInputTemplate = BaseInputTemplate & {
+  nonWitnessUtxo: Buffer;
+};
+
+/**
+ * @description KeyPair to sign the transaction
+ * privateKey is the private key of the sender with unspent utxo. WIF format is expected
+ * network is the network for which the transaction should be encoded.
+ * We use bitcoinjs-lib network types
+ */
+type KeyPair = {
+  privateKey: string;
+  network: Network;
+};
+
+type BaseBitcoinTransferRequest = {
+  output: BaseOutputTemplate;
+  keypair: KeyPair;
+};
+
+/**
+ * @description type for bitcoin witness transaction
+ * keypair is used here to sing the transaction inputUtxo being used
+ */
+type BitcoinWitnessTransferRequest = BaseBitcoinTransferRequest & {
+  inputUtxo: WitnessInputTemplate;
+};
+
+/**
+ * @description type for bitcoin non-witness transaction (Legacy)
+ * keypair is used here to sing the transaction inputUtxo being used
+ */
+type BitcoinNonWitnessTransferRequest = BaseBitcoinTransferRequest & {
+  inputUtxo: NonWitnessInputTemplate;
+};
+
+export function createBitcoinWitnessFungibleTransfer(): BitcoinFungibleAssetTransfer {
+  throw new Error("Method not implemented");
+}
+
+export function createBitcoinNonWitnessFungibleTransfer(): BitcoinFungibleAssetTransfer {
+  throw new Error("Method not implemented");
+}
+
+export abstract class BitcoinFungibleAssetTransfer {
```

Comment:
this should have methods:
```
getRawUtxoRequest() //generated utxo
getUriEncodedUtxoRequest() //bip21 encoded utxo
//maybe even methog to get svg for QR code
```

Reply:
@mpetrunic `getRawUtxoRequest` you mean the raw unsgined transaction?

Reply:
as per the discussion on the call, we won't create transactions, we need to encode transaction params into the bip21 URI so that it can be picked up by bitcoin wallets. Ideally you would install bitcoin wallet and test it out

Reply:
in that case, do we need this `getRawUtxoRequest` method? I'm seeking here clarification of what do you mean by raw utxo request, since json format of outputs have value, script, address and the scriptPubkey atrributes 🤔

Reply:
So I would say we have two methods:
- getRawUtxoRequest() - returns object with information on deposit (I would change the naming of this methods as it is not some standard raw format of transaction)
- getUriEncodedUtxoRequest() - we use bip21 as described

Reply:
@MakMuftic I think for the first function -`getRawUtxoRequest`- , we could return the same parameters that are being passed into the constructor of the abstract class, since we don't need anything else. My only question here is if we should return also as part of the object, the taproot address where all deposits should go?

Reply:
I'm going to update the type